apitools/gen/util.py

```diff
@@ -373,7 +388,16 @@ def FetchDiscoveryDoc(discovery_url, retries=5):
     for url in discovery_urls:
         for _ in range(retries):
             try:
-                content = urllib_request.urlopen(url).read()
+                response = urllib_request.urlopen(url)
+                encoding = response.info().get('Content-Encoding')
+                if encoding == 'gzip':
+                  content = _Gunzip(response.read())
+                elif encoding == 'deflate':
+                  content = response.read()
+                else:
+                  logging.warning('Unknown encoding')
```

Comment:
We shouldn't log a warning if the encoding is not specified. `Content-Encoding` looks like an optional header.

https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.11

Reply:
Done.